packages/communication-react/review/communication-react.api.md

```diff
@@ -1704,6 +1704,7 @@ export type MessageThreadProps = {
     onRenderJumpToNewMessageButton?: (newMessageButtonProps: JumpToNewMessageButtonProps) => JSX.Element;
     onLoadPreviousChatMessages?: (messagesToLoad: number) => Promise<boolean>;
     onRenderMessage?: (messageProps: MessageProps, messageRenderer?: MessageRenderer) => JSX.Element;
+    onRenderFileDownloads?: (userId: string, message: ChatMessage) => JSX.Element;
```

Comment:
# Same design concern as for upload

Symmetric comment to my comment on the `SendBox` API extension -- this new render function is not targeted enough, and does not provide a default implementation for Contoso.

* Contoso could use this to essentially inject anything (it would be better named `onRenderMessageSuffix`)
* By default, file download doesn't work for Contoso. In order to implement it, they have to implement a lot of logic that is internal to `packages/react-composites`.

The alternative API would be:

* MessageThread gains a new prop that is `FileDownloadHandler`
  * MessageThread internally uses `FileDownloadCards`, using `FileDownloadHandler` if it is provided.

Additionally, there could be an `onRenderFileDownloads` prop as here, but it would *override* this default behavior.

@anjulgarg here as well.

Reply:
Moved this discussion to team channel. Let's unblock this PR because this concern is in the same vein as a discussion we already have for `SendBox`. We need to revisit this API a bit.